msdk-io/msdk-io-mztab/src/test/java/io/github/msdk/io/mztab/MzTabFileExportMethodTest.java

```diff
@@ -0,0 +1,166 @@
+
+        // The table has 7 samples
+        List<Sample> samples = featureTable2.getSamples();
+        Assert.assertNotNull(samples);
+        Assert.assertEquals(7, samples.size());
+
+        // The table has columns
+        Assert.assertFalse(featureTable2.getColumns().isEmpty());
+
+        // The table has 298 rows
+        Assert.assertFalse(featureTable2.getRows().isEmpty());
+        Assert.assertEquals(298, featureTable2.getRows().size());
+
+        // ********************
+        // Annotation 7 - HEPES
+        // ********************
+        FeatureTableRow row = featureTable2.getRows().get(7);
+        FeatureTableColumn<List<IonAnnotation>> ionAnnotationColumn = featureTable2
+                .getColumn(ColumnName.IONANNOTATION, null);
+        List<IonAnnotation> ionAnnotations = row.getData(ionAnnotationColumn);
+        Assert.assertNotNull(ionAnnotations);
+        IonAnnotation ionAnnotation = ionAnnotations.get(0);
+        Assert.assertEquals("HEPES", ionAnnotation.getDescription());
+        IChemObjectBuilder builder = DefaultChemObjectBuilder.getInstance();
+        IMolecularFormula cdkFormula = MolecularFormulaManipulator
+                .getMolecularFormula("C8H18N2O4S", builder);
+        String formula = MolecularFormulaManipulator
+                .getString(ionAnnotation.getFormula());
+        String formula2 = MolecularFormulaManipulator.getString(cdkFormula);
+        Assert.assertTrue(formula.equals(formula2));
+
+        // ********************
+        // Row 5
+        // ********************
+        row = featureTable2.getRows().get(5);
+        FeatureTableColumn<Double> mzColumn = featureTable2
+                .getColumn(ColumnName.MZ, null);
+        Double averageMZ = row.getData(mzColumn);
+        Assert.assertNotNull(averageMZ);
+        Assert.assertEquals(520.334738595145, averageMZ, 0.0000001);
+        Sample sample = featureTable2.getSamples().get(5);
+        Assert.assertEquals("36C sample 2", sample.getName());
+
+        // ********************
+        // Last row
+        // ********************
+        row = featureTable2.getRows().get(297);
+        sample = featureTable2.getSamples().get(5);
+
+        // Area
+        FeatureTableColumn<Double> areaColumn = featureTable2
+                .getColumn(ColumnName.AREA, sample);
+        Double area = row.getData(areaColumn);
+        Assert.assertNotNull(area);
+        Assert.assertEquals(11480605.3259747, area, 0.0000001);
+
+        // RT
+        FeatureTableColumn<ChromatographyInfo> chromInfoColumn = featureTable2
+                .getColumn("Chromatography Info", null,
+                        ChromatographyInfo.class);
+        ChromatographyInfo chromatographyInfo = row.getData(chromInfoColumn);
+        Assert.assertNotNull(chromatographyInfo);
+        Float rt = chromatographyInfo.getRetentionTime();
+        Assert.assertNotNull(rt);
+        Assert.assertEquals(30.324697494506836, rt, 0.0000001);

```

Comment:
After doing some more debugging, I found that the error was caused because `sm.setRetentionTime(rtValue);` was run before `sm.setOptionColumnValue(new Assay(sampleCounter), "rt", rtValue);`. This resulted in the common RT value being set to the value of the last sample. The order has therefore been changed in #112.